include/async_scope.hpp

```diff
@@ -708,180 +364,283 @@ namespace _P2519::execution {
         std::unique_ptr<__future_state<_Sender>> __state_;
       };
 
-    struct async_scope {
-    private:
-      struct __end_of_scope_callback {
-          async_scope* __async_scope_;
-          void operator()() {
-            __async_scope_->__end_of_scope_(); // enter stop state
-          }
-      };
-
-      in_place_stop_source __stop_source_;
-      in_place_stop_callback<__end_of_scope_callback> __end_of_scope_callback_;
-
-      // (__op_state_ & 1) is 1 until we've been stopped
-      // (__op_state_ >> 1) is the number of outstanding operations
-      std::atomic<std::size_t> __op_state_{1};
-      __async_manual_reset_event __evt_;
-
-      struct __load_atomic {
-        const std::atomic<std::size_t>& __op_state_;
-        void operator()() noexcept {
-          // make sure to synchronize with all the fetch_subs being done while
-          // operations complete
-          (void) __op_state_.load(std::memory_order_acquire);
+    struct async_scope : __immovable {
+        ~async_scope() noexcept {
+            std::unique_lock __guard{__lock_};
+            if (__active_ != 0) { 
+              std::terminate();
+            }
+            if (!__waiters_.empty()) { 
+              std::terminate();
+            }
         }
-      };
-
-      [[nodiscard]] auto __await_and_sync_() const noexcept {
-        return then(__evt_.async_wait(), __load_atomic{__op_state_});
-      }
+        async_scope() 
+        : __active_(0) {}
 
-    public:
-      async_scope() noexcept
-        : __end_of_scope_callback_(__stop_source_.get_token(), __end_of_scope_callback{this})
-        , __evt_(true)
-      {}
-
-      ~async_scope() {
-        __end_of_scope_(); // enter stop state
+        struct __op_base : __immovable {
+            explicit __op_base(async_scope* __scope) : __scope_(__scope), __next_(nullptr) {}
+            async_scope* __scope_;
+            __op_base* __next_;
+        protected:
+            virtual void notify_waiter() noexcept =0;
+            friend void __empty_sender_notify_waiter(__op_base& __self) {
+                __self.notify_waiter();
+            }
+        };
+        // a constraint that starts a supplied sender when there are no active senders
+        struct __when_empty {
+        template<class __ConstrainedId>
```

Comment:
needs an indent

Reply:
is this something that clang-format can automate?